lms/db/__init__.py

```diff
@@ -88,6 +89,13 @@ def __repr__(self):
 SESSION = sessionmaker()
 
 
+class TimestampedModelMixin:
+    created_at = sqlalchemy.Column(sqlalchemy.DateTime, default=func.now())
+    updated_at = sqlalchemy.Column(
+        sqlalchemy.DateTime(timezone=True), onupdate=func.now()
+    )
```

Comment:
A mixin class seems like a good idea for this :+1: We have a similar thing in h: https://github.com/hypothesis/h/blob/5101ff6045f76b64f9358d366e68e4cbf6c64a3b/h/db/mixins.py#L8-L21 And in Checkmate (though not for timestamps): https://github.com/hypothesis/checkmate/blob/main/checkmate/models/db/mixins.py

Might be nicer in `models/_mixins.py` rather than in `db`.

Are there any existing models for LMS that could be changed to use this mixin?

I'm not sure what the correct / best way to do timestamps in the DB is:

* Should they have `timezone=True`?

* Why does `updated_at` have `timezone=True` but not `created_at`?

* I think `created_at` should probably have `server_default=sa.func.now()` rather than `default`, and `updated_at` should also have `server_default=sa.func.now()`, that way the user of a model class doesn't have to do anything about `created_at` and `updated_at` they just get set to the right values automatically.

  (Generally I think you're supposed to use `server_default` rather than `default` in SQLAlchemy unless you have a reason not to. `server_default` actually sets the default in the Postgres schema so direct SQL that doesn't go through the models will also use the default.)

* Should it be `sa.func.now()` or `sa.func.utcnow()`?

* They should probably have `nullable=False`, no reason these should ever be null

Reply:
I'll do the Mixin in a different PR :+1: 
https://github.com/hypothesis/lms/pull/2622